quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -226,6 +227,23 @@ instance IsTest QC where
       (qcOutputNl ++
         (if putReplayInDesc then replayMsg else ""))
 
+
+-- | Like the original 'QC.quickCheck' but is reporting progress using tasty
+-- callback.
+--
+quickCheck :: (Progress -> IO ())
+           -> QC.Args
+           -> QC.Property
+           -> IO QC.Result
+quickCheck yieldProgress args prop = do
+  -- Here we rely on the fact that QuickCheck currently prints its progress to
+  -- stderr and the overall status (which we don't need) to stdout
+  tm <- QC.newTerminal
+          (const $ pure ())
+          (\progressText -> yieldProgress emptyProgress { progressText })
```

Comment:
Can we extract `progressPercent` somehow?

Reply:
We'd need to parse what quickcheck provides ([ref](https://hackage.haskell.org/package/QuickCheck-2.14.2/docs/src/Test.QuickCheck.Test.html#showTestCount)), but this would work until `quickcheck` changes the string format.

Reply:
Let's extract the digits.  Unfortunately QuickCheck doesn't provide a method to extract `maxSuccess` for individual tests, if it did we could show the percentage of progress.

Reply:
Done.